### PR TITLE
Listing the data sources

### PR DESCRIPTION
**Pull request by alfretoine**
### **- The menu**

Screenshot 2021-01-19 at 15 09 18

### **- The listing of data sources**

Screenshot 2021-01-19 at 15 08 45

### CODE REVIEW

**Comment by madewulf** [comment]
LGTM , I tested it on local DB, works well. It just seems to me that we have to clean up the authorization screen, now, but this can be another PR

Screenshot 2021-01-21 at 12 11 30

**Comment by madewulf** [comment]
Screenshot 2021-01-21 at 12 13 46
JS Tests are failing, though

File: hat/assets/js/apps/Iaso/domains/dataSources/index.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+const styles = theme => ({
+    ...commonStyles(theme),
+    reactTable: {
+        ...commonStyles(theme).reactTable,
+        marginTop: theme.spacing(4),
+    },
+});
+
+class DataSources extends Component {
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/dataSources/index.js:31]: This one can be clearly updated to a functional component with hooks, then avoid repetition of the code.

File: hat/assets/js/apps/Iaso/domains/dataSources/index.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        return (
+            <>
+                {fetching && <LoadingSpinner />}
+                <TopBar
+                    title={formatMessage(MESSAGES.dataSources)}
+                    displayBackButton={false}
+                />
+                <Box className={classes.containerFullHeightNoTabPadded}>
+                    <div className={classes.reactTable}>
+                        <CustomTableComponent
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/dataSources/index.js:79]: Again not mandatory but try to use TableComponent.js instead ;-)

File: hat/assets/js/apps/Iaso/domains/dataSources/index.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    intl: PropTypes.object.isRequired,
+    params: PropTypes.object.isRequired,
+    fetchAllDataSources: PropTypes.func.isRequired,
+    fetching: PropTypes.bool.isRequired,
+    reduxPage: PropTypes.object,
+};
+
+const MapStateToProps = state => ({
+    fetching:
+        state.dataSources !== undefined &&
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/dataSources/index.js:118]: lol, probably my bad but this logic has to be in the reducer no ?

**Review by beygorghor — APPROVED** [review]
Thx for the tests !

**Pull request merged**